core/pkg/utils/build_notifications.go

```diff
@@ -0,0 +1,46 @@
+package utils
```

Comment:
This was extracted from the store, since we also need the same logic [here](https://github.com/open-feature/flagd/pull/1702/files#diff-23c9326d5b7c4e77c939963ea1010025ca5faf16eb9db96b348d7c19cb921fbeR41) now, to diff flag changes on the RPC event stream.